### PR TITLE
Replace `--master-az` with `--control-plane-az` in `template nodepool`

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/14964

This changes the `kuebctl gs template cluster` command to deprecate the flag `--master-az` and introduce `--control-plane-az` instead.

### CODE REVIEW

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -19,13 +19,14 @@ const (
...
-	flagClusterID = "cluster-id"
-	flagMasterAZ  = "master-az"
-	flagName      = "name"
-	flagOutput    = "output"
-	flagOwner     = "owner"
-	flagRelease   = "release"
-	flagLabel     = "label"
+	flagClusterID      = "cluster-id"
+	flagControlPlaneAZ = "control-plane-az"
+	flagMasterAZ       = "master-az" // TODO: Remove some time after August 2020
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:24]:
```suggestion
	flagMasterAZ       = "master-az" // TODO: Remove some time after August 2021
```

File: cmd/template/cluster/flag.go [hunk 2]
```diff
@@ -54,17 +56,26 @@ func (f *flag) Init(cmd *cobra.Command) {
...
-	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, []string{}, "Workload master node availability zone.")
+	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, []string{}, "Availability zone(s) to use by control plane nodes.")
+	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, []string{}, "Replaced by --control-plane-az.")
 	cmd.Flags().StringVar(&f.Name, flagName, "", "Workload cluster name.")
 	cmd.Flags().StringVar(&f.Output, flagOutput, "", "File path for storing CRs.")
 	cmd.Flags().StringVar(&f.Owner, flagOwner, "", "Workload cluster owner organization.")
 	cmd.Flags().StringVar(&f.Release, flagRelease, "", "Workload cluster release. If not given, this remains empty for defaulting to the most recent one via the Management API.")
 	cmd.Flags().StringSliceVar(&f.Label, flagLabel, nil, "Workload cluster label.")
+
+	// TODO: Remove the flag completely some time after August 2020
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:67]:
```suggestion
	// TODO: Remove the flag completely some time after August 2021
```

File: cmd/template/cluster/flag.go [hunk 3]
```diff
@@ -54,17 +56,26 @@ func (f *flag) Init(cmd *cobra.Command) {
 
 	// Common.
 	cmd.Flags().StringVar(&f.ClusterID, flagClusterID, "", "User-defined cluster ID.")
-	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, []string{}, "Workload master node availability zone.")
+	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, []string{}, "Availability zone(s) to use by control plane nodes.")
+	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, []string{}, "Replaced by --control-plane-az.")
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:60]:
```suggestion
	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, nil, "Availability zone(s) to use by control plane nodes.")
	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, nil, "Replaced by --control-plane-az.")
```

**Review by axbarsan — COMMENTED** [review]
Why remove all the logic regarding the `--master-az` flag? If an user sets it, nothing would change in the generated CR.

**Comment by marians** [comment]
The last commit fixes the behaviour. The deprecated flag should in fact still be usable. I just forgot to assign its value.

File: cmd/template/cluster/flag.go [hunk 4]
```diff
@@ -54,17 +56,26 @@ func (f *flag) Init(cmd *cobra.Command) {
 
 	// Common.
 	cmd.Flags().StringVar(&f.ClusterID, flagClusterID, "", "User-defined cluster ID.")
-	cmd.Flags().StringSliceVar(&f.MasterAZ, flagMasterAZ, []string{}, "Workload master node availability zone.")
+	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, []string{}, "Availability zone(s) to use by control plane nodes.")
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:59]:
```suggestion
	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, nil, "Availability zone(s) to use by control plane nodes.")
```

File: cmd/template/cluster/flag.go [hunk 5]
```diff
@@ -110,12 +121,12 @@ func (f *flag) Validate() error {
 		// Validate Master AZs.
 		switch f.Provider {
 		case key.ProviderAWS:
-			if len(f.MasterAZ) != 0 && len(f.MasterAZ) != 1 && len(f.MasterAZ) != 3 {
-				return microerror.Maskf(invalidFlagError, "--%s must be set to either one or three availability zone names", flagMasterAZ)
+			if len(f.ControlPlaneAZ) != 0 && len(f.ControlPlaneAZ) != 1 && len(f.ControlPlaneAZ) != 3 {
+				return microerror.Maskf(invalidFlagError, "--%s must be set to either one or three availability zone names", flagControlPlaneAZ)
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:125]: I believe we need to also validate `f.MasterAZ` with the same rules

Comment (marians) [reply cmd/template/cluster/flag.go:140]: The recent commit simply copies the value of f.MasterAZ to f.ControlPlaneAZ, so validation for f.MasterAZ isn't necessary.

**Review by axbarsan — APPROVED** [review]

**Pull request merged**